h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -99,6 +100,16 @@ public void initMetalearnerParams(Metalearner.Algorithm algo) {
     
     public final Frame blending() { return _blending == null ? null : _blending.get(); }
 
+    @Override
+    public String[] getNonPredictors() {
+      HashSet<String> nonPredictors = new HashSet<>();
+      nonPredictors.addAll(Arrays.asList(super.getNonPredictors()));
+      if (null != _metalearner_parameters)
+        nonPredictors.addAll(Arrays.asList(_metalearner_parameters.getNonPredictors()));
+      if (null != _metalearner_fold_column)
+        nonPredictors.add(_metalearner_fold_column);
+      return nonPredictors.toArray(new String[0]);
+    }
```

Comment:
so you're using my hint for another PR in this one? I first got confused when I saw this change :)

I have a doubt about the necessity to include
```
      if (null != _metalearner_parameters)
        nonPredictors.addAll(Arrays.asList(_metalearner_parameters.getNonPredictors()));
```
can you please justify?

Reply:
Yes, sorry, I usually read through the comments and often I find something new so I want to use it.

My assumption is that there can be a metalearner that would use a different column as a non predictor which would be otherwise removed.
 
For example, I am not sure if HGLM's `random_columns`[1] are included in the predictors so I delegated the responsibility to GLM to decide what's needed. 

(This assumes that the individual algos override the `getNonPredictors` when needed.)

I know this is poor justification, I will look at it on Monday to make sure if there actually some use case for that if condition.

[1] https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/algo-params/random_columns.html

Reply:
Even if your justification makes sense for edge cases, I don't think we support those anyway as the training frame given to the metalearner will contain only the predictions columns from the base models + the ones listed in `StackedEnsemble.addMiscColumnsToLevelOneFrame`.

For clarity, I'm just expecting that the `non-predictors` are exactly those in `addMiscColumnsToLevelOneFrame`.

Reply:
OK, I removed the condition and renamed `addMiscColumnsToLevelOneFrame` to `addNonPredictorsToLevelOneFrame` make its purpose more clear.